connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/webhook/InboundWebhookRestController.java

```diff
@@ -201,6 +214,24 @@ private ResponseEntity<?> buildErrorResponse(CorrelationResult.Failure failure)
     ResponseEntity<?> response;
     if (failure instanceof CorrelationResult.Failure.Other) {
       response = ResponseEntity.internalServerError().build();
+    } else if (failure instanceof CorrelationResult.Failure.ZeebeClientStatus zeebeClientStatus) {
+      response =
+          switch (zeebeClientStatus.status()) {
+            case "CANCELLED" -> ResponseEntity.status(499).body(failure);
+            case "UNKNOWN", "INTERNAL", "DATA_LOSS" -> ResponseEntity.status(500).body(failure);
+            case "INVALID_ARGUMENT" -> ResponseEntity.status(400).body(failure);
+            case "DEADLINE_EXCEEDED" -> ResponseEntity.status(504).body(failure);
+            case "NOT_FOUND" -> ResponseEntity.status(404).body(failure);
+            case "ALREADY_EXISTS", "ABORTED" -> ResponseEntity.status(409).body(failure);
+            case "PERMISSION_DENIED" -> ResponseEntity.status(403).body(failure);
+            case "RESOURCE_EXHAUSTED" -> ResponseEntity.status(429).body(failure);
+            case "FAILED_PRECONDITION" -> ResponseEntity.status(412).body(failure);
```

Comment:
You could also try to "parse" the code and do the switch based on the enum. This would make it more safe would lead to a compilation error in case we did not handle the error. 
On main we could also change the API and return the actual zeebe status.